eox_core/api/support/v1/tests/integration/test_views.py

```diff
@@ -0,0 +1,354 @@
+        self.create_user(self.tenant_x, data)
+
+        response = self.delete_user(self.tenant_x, {query_param: data[query_param]})
+        response_data = response.json()
+        get_response = self.get_user(self.tenant_x, {"email": data["email"]})
+        get_response_data = get_response.json()
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response_data, f"The user {data['username']} <{data['email']}>  has been removed")
+        self.assertEqual(get_response.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertEqual(
+            get_response_data,
+            {"detail": f"No user found by {{'email': '{data['email']}'}} on site {self.tenant_x['domain']}."},
+        )
+
+    @ddt.data(
+        ("username", "user-not-found"),
+        ("email", "user-not-found@mail.com"),
+    )
+    @ddt.unpack
+    def test_delete_user_in_tenant_not_found(self, query_param: str, value: str) -> None:
+        """
+        Test delete an edxapp user in a tenant that does not exist.
+
+        Open edX definitions tested:
+        - `get_edxapp_user`
+
+        Expected result:
+        - The status code is 404.
+        - The response indicates the user was not found in the tenant.
+        """
+        response = self.delete_user(self.tenant_x, {query_param: value})
+        response_data = response.json()
+
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertEqual(
+            response_data["detail"],
+            f"No user found by {{'{query_param}': '{value}'}} on site {self.tenant_x['domain']}.",
+        )
+
+    @ddt.data("username", "email")
+    def test_delete_user_of_another_tenant(self, query_param: str) -> None:
+        """
+        Test delete an edxapp user of another tenant.
+
+        Open edX definitions tested:
+        - `get_edxapp_user`
+
+        Expected result:
+        - The status code is 404.
+        - The response indicates the user was not found in the tenant.
+        """
+        data = next(FAKE_USER_DATA)
+        self.create_user(self.tenant_x, data)
+
+        response = self.delete_user(self.tenant_y, {query_param: data[query_param]})
+        response_data = response.json()
+
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertEqual(
+            response_data["detail"],
+            f"No user found by {{'{query_param}': '{data[query_param]}'}} on site {self.tenant_y['domain']}.",
+        )
+
+    def test_delete_user_missing_required_fields(self) -> None:
+        """
+        Test delete an edxapp user in a tenant without providing the username or email.
+
+        Expected result:
+        - The status code is 400.
+        - The response indicates the username or email is required.
+        """
+        response = self.delete_user(self.tenant_x)
+        response_data = response.json()
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
```

Comment:
I don't think it's necessary to check it here, anyway in those tests no users are being created.

Reply:
I understand that we aren’t creating any users in these tests, but is it possible that users created in previous tests might still exist? If I’m not mistaken, these tests don’t reset the database after each test case, so any user created would persist unless explicitly deleted. That’s why I think it could make sense to ensure our delete operation doesn’t unintentionally remove any users.

That said, this type of check might not be necessary, as it’s not directly testing an integration scenario. Still, it’s something we can think about further to define the boundaries of database state verification. For now, though, I think it’s fine as is

Reply:
@magajh: Now that you mention it, should we consider using a teardown method to undo all operations made during the tests in the near future? This way, we know running a test suite is an idempotent operation.